pkg/utils/openshift/utils.go

```diff
@@ -1,37 +1,53 @@
 package openshift
 
 import (
-	"github.com/RHsyseng/operator-utils/pkg/platform"
+	"github.com/RHsyseng/operator-utils/internal/platform"
 	"k8s.io/client-go/rest"
 )
 
 /*
-IsOpenShift tests the Kubernetes-based environment for indicators that the running platform is OpenShift.
+GetPlatformInfo examines the Kubernetes-based environment and determines the running platform, version, & OS.
 Accepts <nil> or instantiated 'cfg' rest config parameter.
+
+Result: PlatformInfo{ Name: OpenShift, K8SVersion: 1.13+, OS: linux/amd64 }
 */
-func IsOpenShift(cfg *rest.Config) (bool, error) {
-	return platform.DetectOpenShift(nil, cfg)
+func GetPlatformInfo(cfg *rest.Config) (platform.PlatformInfo, error) {
+	return platform.K8SBasedPlatformVersioner{}.GetPlatformInfo(nil, cfg)
 }
 
 /*
-GetPlatformInfo examines the Kubernetes-based environment and determines the running platform, version, & OS.
-Returned PlatformInfo.OCPVersion is derived from underlying PlatformInfo.K8SVersion.
+IsOpenShift is a helper method to simplify boolean OCP checks against GetPlatformInfo results
 Accepts <nil> or instantiated 'cfg' rest config parameter.
-
-Result: PlatformInfo{ Name: OpenShift, OCPVersion: 4.1, K8SVersion: 1.13+, OS: linux/amd64 }
 */
-func GetPlatformInfo(cfg *rest.Config) (platform.PlatformInfo, error) {
-	return platform.K8SBasedPlatformVersioner{}.GetPlatformInfo(nil, cfg)
+func IsOpenShift(cfg *rest.Config) (bool, error) {
+	info, err := GetPlatformInfo(cfg)
+	if err != nil {
+		return false, err
+	}
+	return info.IsOpenShift(), nil
 }
 
 /*
-LookupOpenShiftVersion refines PlatformInfo.OCPVersion info based on API calls, rather than deriving from K8SVersion.
+LookupOpenShiftVersion fetches OpenShift version info from API endpoints
 *** NOTE: OCP 4.1+ requires elevated user permissions, see PlatformVersioner for details
 Accepts <nil> or instantiated 'cfg' rest config parameter.
-Accepts prepopulated or empty PlatformInfo{}
 
-Result: PlatformInfo{ Name: OpenShift, OCPVersion: 4.1.2, K8SVersion: 1.13+, OS: linux/amd64 }
+Result: OpenShiftVersion{ Version: 4.1.2 }
+*/
+func LookupOpenShiftVersion(info platform.PlatformInfo) (platform.OpenShiftVersion, error) {
```

Comment:
Without doing a deep dive, I'm surprised you don't need `cfg *rest.Config` here?!

Reply:
You're right, I factored part of it away when there was still some backend overlap with PlatformInfo, but they share no logic now and each need cfg to make their own calls. Forgot to swap it back out - updated now.